app/models/allocation.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+class Allocation < VersionedModel
+  enum prisoner_category: {
+    b: 'B',
+    c: 'C',
+    d: 'D',
+  }
+
+  enum sentence_length: {
+    any: 'any',
+    under_18_months: 'under18',
+    over_18_months: 'over18',
+  }
+
+  belongs_to :from_location, class_name: 'Location'
+  belongs_to :to_location, class_name: 'Location'
+
+  validates :from_location, presence: true
+  validates :to_location, presence: true
+
+  validates :prisoner_category, inclusion: { in: prisoner_categories }
+  validates :sentence_length, inclusion: { in: sentence_lengths }
+
+  validates :moves_count, presence: true
```

Comment:
The naming of this column is deliberate - will likely use `:counter_cache` to populate this in future. For now it will be set explicitly in the front end.